src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,42 +1,73 @@
-import React from 'react';
+import React, { useCallback, useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../hooks';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+import { selectTodo } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const currentTodos = useAppSelector(state => state.currentTodo);
+  const [user, setUser] = useState<User | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    setIsLoading(true);
+    if (currentTodos) {
+      getUser(currentTodos?.userId)
+        .then(setUser)
+        // eslint-disable-next-line no-console
+        .catch(error => console.log(error.message))
+        .finally(() => setIsLoading(false));
+    }
```

Comment:
move the logic inside the useEffect hook to a separate function and move useEffect to the end of the code